www/docs/cookbooks/sign-and-notarizing-osx.md

```diff
@@ -0,0 +1,270 @@
+# .goreleaser.yml
+env:
+  - BUNDLE_ID=com.example.myapp
+  - APPLE_ID_USERNAME=user@example.com
+  - APPLE_ID_PASSWORD=@keychain:gon
+  - "APPLE_APPLICATION_IDENTITY=Developer ID Application: My Name"
+
+builds:
+  - id: default
+    goos: [linux, windows]
+    goarch: [arm64, amd64]
+    ignore:
+      - goarch: arm
+        goos: windows
+      - goarch: arm64
+        goos: windows
+  - id: macos
+    goos: [darwin]
+    goarch: [arm64, amd64]
+    hooks:
+      post:
+        - |
+          sh -c '
+          fn=dist/macos_{{.Target}}/gon.hcl
+          cat >"$fn" <<EOF
+          bundle_id = "{{.Env.BUNDLE_ID}}"
+          apple_id {
+            username = "{{.Env.APPLE_ID_USERNAME}}"
+            password = "{{.Env.APPLE_ID_PASSWORD}}"
+          }
+          source = ["dist/macos_{{.Target}}/{{.Name}}"]
+          sign {
+            application_identity = "{{.Env.APPLE_APPLICATION_IDENTITY}}"
+          }
+          EOF
+          '
+        - "gon 'dist/macos_{{.Target}}/gon.hcl'"
+```
+
+Result: the `dist/macos_*/myapp` will be signed.
+
+```
+$ codesign --verify --deep --verbose dist/macos_darwin_arm64/myapp
+dist/macos_darwin_arm64/myapp: valid on disk
+dist/macos_darwin_arm64/myapp: satisfies its Designated Requirement
+```
+
+
+## Configuring Notarization
+
+Notarizing takes our archived (with a signed binary inside) and submits it to Apple. This is an online activity than can take meaningful time but it necessary for production releases to avoid security prompts.
+
+Here we are doing a very similar behavior as we did during the Build step, where we define a hon config block on the fly and then execute.
+
+We must be sure to define an archive ID, so we when we go to Release later, Goreleaser will pick the correct files. Here we do a specific notarize step _replacing_ the original artifact in-place. One could generate a separate archive filename, just be sure not to Release it.
+
+**Important**:
+  - zip, pkg, dmg, app format is required for notarization.
+
+```yaml
+# .goreleaser.yml
+
+archives:
+  - format: zip
+    builds: [default]
+    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
+    format_overrides:
+      - goos: linux
+        format: tar.gz
+  - id: macos
+    builds: [macos]
+    format: zip
+    name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
+
+signs:
+  - id: mac-notarize
+    artifacts: archive
+    ids: [macos]
+    signature: "${artifact}"
+    output: true
+    cmd: sh
+    args:
+      - "-c"
+      - |-
+        cat >"dist/gon.notarize.hcl" <<EOF
+        apple_id {
+          username = "{{.Env.APPLE_ID_USERNAME}}"
+          password = "{{.Env.APPLE_ID_PASSWORD}}"
+        }
+        notarize {
+          path = "${artifact}"
+          bundle_id = "{{.Env.BUNDLE_ID}}"
+        }
+        EOF
+        gon "dist/gon.notarize.hcl"
+```
+
+Result: notarized artifacts.
+FIXME: We know it notarized correctly, why is our archive being rejected by spctl?
```

Comment:
I'm not sure if it would expect we to notarize the archive or the actual binary... maybe the issue lies there?

Reply:
Signing happens on the binary, notarization happens on the (compatible) archive. Notarizing without first signing the archive's binary results in an error.

Reply:
I've been tempted to play with integrating hon as a library, but it's this particular requirement that I'm struggling with. Ideally we only want to sign at the sign step, but that'd require us to change the built in binary in the archive so that it's signed, and then submit the entire archive for notarization. Al far as I can tell, that doesn't really fit with goreleaser's model.